src/main/java/org/commcare/formplayer/beans/menus/QueryResponseBean.java

```diff
@@ -69,6 +68,14 @@ public QueryResponseBean(QueryScreen queryScreen, SessionWrapper session) {
                 choiceLabels = ItemSetUtils.getChoiceLabels(queryPromptItem.getItemsetBinding());
             }
 
+            String requiredMessage = null;
+            QueryPromptCondition required = queryPromptItem.getRequired();
+            if (required != null && required.getMessage() !=null) {
+                requiredMessage = queryPromptItem.getRequired().getMessage().evaluate(
+                        session.getEvaluationContext());
+            }
```

Comment:
I think this could also be pushed down:
```java
String requiredMessage = queryPromptItem.getRequiredMessage(ec);
```

Another option would be to make the 'getRequiredPrompts` include the message:

```java
Hashtable<String, String> requiredPrompts = queryScreen.getRequiredPromptsWithMessages();
```

The presence of a key in the hashtable indicates that the prompt is required while the value is the message (which could be null).